src/overlay.rs

```diff
@@ -112,11 +112,17 @@ where
     ///
     /// The overlay will then point at the newly written state and tree version.
     #[instrument(name = "WriteOverlay::commit", skip(self, writer))]
-    pub async fn commit<W>(&mut self, mut writer: W) -> Result<(RootHash, Version)>
+    pub async fn commit<W>(&mut self, mut writer: W) -> Result<Option<(RootHash, Version)>>
     where
         W: TreeWriter + Sync,
     {
-        let overlay = std::mem::replace(&mut self.overlay, Default::default());
+        // If there's nothing to write, we should immediately return `None` without doing anything.
+        // If we don't do this check, then `put_value_set` will panic.
+        if self.overlay.is_empty() {
+            return Ok(None);
+        }
```

Comment:
This was necessary to add in order to prevent a `commit` on an unchanged `WriteOverlay` causing a panic.